tests/packages/vendored-meson/meson.build

```diff
@@ -0,0 +1,13 @@
+# SPDX-FileCopyrightText: 2023 The meson-python developers
+#
+# SPDX-License-Identifier: MIT
+
+project('vendored-meson', version: '1.0.0')
+
+py = import('python').find_installation()
+
+py.install_sources('pure.py')
+
+if get_option('custom-cli-used')
+    py.install_sources('extra/pure.py', subdir: 'extra')
+endif
```

Comment:
I think we could avoid having to build the wheel for the test using the Meson's `error()` function:
```meson
if not get_option('custom-cli-used')
    error('Expected option "custom-cli-used" was not specified')
endif
```
then the test package can be empty except for the meson wrapper script.